### PR TITLE
Added a functional private secret manager example

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

A private only secret manager example was added to this repo. All endpoint_types have been set to private for this example. It has been tested in schematics.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/private/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Basic example
```
Comment (ocofaigh) [inline examples/private/README.md]: Private example

Also update the content below since its a private only example

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Can you update `.catalog-onboard-pipeline.yaml` to add the new example to it, and also update the [flavors](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager-iam-engine/blob/880ebcc8bc7f8b32a65553669861afb61961c1fb/ibm_catalog.json#L47) array in the ibm_catalog.json with the new example too

File: examples/private/README.md (hunk 1 above)
Comment (kierramarie) [reply examples/private/README.md]: Let me know if there is anything else you'd like me to add to the docs.

File: ibm_catalog.json [hunk 2]
```diff
@@ -61,6 +61,11 @@
               }
             ]
           }
+        },
+        {
+          "label": "Private Only",
+          "name": "private",
+          "working_directory": "examples/private"
```
Comment (ocofaigh) [inline ibm_catalog.json]: you need to also include and `architecture` section. Just copy same format as basic example

File: ibm_catalog.json [hunk 3]
```diff
@@ -61,6 +61,11 @@
               }
             ]
           }
+        },
+        {
+          "label": "Private Only",
```
Comment (ocofaigh) [inline ibm_catalog.json]: Maybe just call it `Private` ?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see latest comments

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**